diff --git a/lib/carbon/conf.py b/lib/carbon/conf.py
index 4bcc113..e955fa8 100644
--- a/lib/carbon/conf.py
+++ b/lib/carbon/conf.py
@@ -402,6 +402,9 @@ def get_default_parser(usage="%prog [options] <start|stop|status>"):
         "--profile",
         help="Record performance profile data to the given file")
     parser.add_option(
+        "--prefix",
+        help="Prefix tag for direct to syslog logging")
+    parser.add_option(
         "--pidfile", default=None,
         help="Write pid to the given file")
     parser.add_option(
@@ -427,6 +430,9 @@ def get_default_parser(usage="%prog [options] <start|stop|status>"):
         "--instance",
         default='a',
         help="Manage a specific carbon instance")
+    parser.add_option(
+        "--syslog", action="store_true",
+        help="Log directly to syslog")
 
     return parser
 
diff --git a/lib/carbon/util.py b/lib/carbon/util.py
index 3ef2e1d..84f293b 100644
--- a/lib/carbon/util.py
+++ b/lib/carbon/util.py
@@ -18,7 +18,7 @@ except:
 from time import sleep, time
 from twisted.python.util import initgroups
 from twisted.scripts.twistd import runApp
-from twisted.scripts._twistd_unix import daemonize
+import daemonize
 
 
 daemonize = daemonize  # Backwards compatibility
@@ -72,8 +72,12 @@ def run_twistd_plugin(filename):
         twistd_options.append("--profile")
     if options.pidfile:
         twistd_options.extend(["--pidfile", options.pidfile])
+    if options.prefix:
+        twistd_options.extend(["--prefix", options.prefix])
     if options.umask:
         twistd_options.extend(["--umask", options.umask])
+    if options.syslog:
+        twistd_options.append("--syslog")
 
     # Now for the plugin-specific options.
     twistd_options.append(program)
@@ -83,7 +87,7 @@ def run_twistd_plugin(filename):
 
     for option_name, option_value in vars(options).items():
         if (option_value is not None and
-            option_name not in ("debug", "profile", "pidfile", "umask")):
+            option_name not in ("debug", "profile", "prefix", "pidfile", "syslog", "umask")):
             twistd_options.extend(["--%s" % option_name.replace("_", "-"),
                                    option_value])
 
